client/state/selectors/eligible-for-free-to-paid-upsell.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	canCurrentUser,
+	isMappedDomainSite,
+	isSiteOnFreePlan,
+	isUserRegistrationDaysWithinRange,
+} from 'state/selectors/';
+
+/**
+ * Returns true if the current user is eligible to participate in the free to paid plan upsell for the site
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  moment Current moment for determination of elapsed days since registration
+ * @return {bool}           True if the user can participate in the free to paid upsell
+ */
+const eligibleForFreeToPaidUpsell = ( state, siteId, moment ) => {
```

Comment:
Yes - the free_plan specifically includes only WordPress.com free plan - jetpack plans have a different product slug.